### ISSUE TITLE
Failing Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests

### ISSUE THREAD

**Issue by natidea** [dotnet/roslyn#8807]
Affected PRs
- https://github.com/dotnet/roslyn/pull/8802
- https://github.com/dotnet/roslyn/pull/8796
- https://github.com/dotnet/roslyn/pull/8812
- https://github.com/dotnet/roslyn/pull/8801
- https://github.com/dotnet/roslyn/pull/8799
- https://github.com/dotnet/roslyn/pull/8646
- https://github.com/dotnet/roslyn/pull/8435

Test Result (12 failures / +12)
- Microsoft.CodeAnalysis.CSharp.UnitTests.DocumentationModeTests.XmlSyntaxError_Included
- Microsoft.CodeAnalysis.CSharp.UnitTests.DocumentationCommentCompilerTests.IncludedName_DuplicateNameAttribute
- Microsoft.CodeAnalysis.CSharp.UnitTests.DocumentationCommentCompilerTests.PragmaDisableWarningInXmlFile

- Microsoft.CodeAnalysis.CSharp.UnitTests.DocumentationCommentCompilerTests.WRN_XMLParseIncludeError_Source
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_XDocument_WRN_XMLDocInvalidXMLFragment
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_XmlError_WRN_XMLDocBadFormedXML
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Bug684184
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_IOError_WRN_XMLDocBadFormedXML
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_FileNotFound_WRN_XMLDocBadFormedXML
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_XPathNotFound_WRN_XMLDocInvalidXMLFragment
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_Cycle_WRN_XMLDocInvalidXMLFragment
- Microsoft.CodeAnalysis.VisualBasic.UnitTests.DocCommentTests.Include_XPathError_WRN_XMLDocBadFormedXML

**Comment by natidea** [comment]
@dotnet/roslyn-compiler These tests are failing consistently, so not really flaky. May have regressed from a recent change?

**Comment by genlu** [comment]
I tried the tests on my machine, couldn't repro.

**Comment by davkean** [comment]
Make note that the --- in the path the new path shortener extension on Jenkins. Perhaps these interprete "-" as having meaning?

**Comment by mmitche** [comment]
In Jenkins the job name and folders are integrated into the paths that the repo gets cloned into.  This is fine for most cases, but on Windows if we have a long-ish job name and a long set of repo paths, we can easily overrun the MAX_PATH.  Git won't even clone in certain cases.  The shortener detects if the workspace path length doesn't leave enough room, abbreviates the base path (with ---) and appends a substring of the digest of the full path to the end of it.  That's why you're getting the -.

**Comment by natidea** [comment]
Some of the failures look like:
```
Expected: ···space\roslyn_prtest---345e02d2\Binaries\Temp\RoslynTests\Docu··· 
Actual: ···space\roslyn_prtest- - -345e02d2\Binaries\Temp\RoslynTests\Docu··· 
```
So the test already expects to see the dashes but for some reason, it changed from the expected `---` to `- - -` with spaces

**Comment by cston** [comment]
The "---" in the path is being replaced when the path is used in an `XComment`. I'm investigating a fix.

**Comment by cston** [comment]
The XML generated by the compilers for doc comments includes references to included files in cases where the included files are invalid. Those references are in XML comments. Since comments cannot include "--", any "--" in the file path is serialized by the `XComment` as "- -".

For instance ```/// <include file="--.xml"/>``` results in:
```
<member>
  <!-- Badly formed XML file "- -.xml" cannot be included -->
</member>
```

By contrast the legacy compilers add a comment along with the original ```<include/>``` in error cases.
```
<member>
  <!-- Failed to insert some or all of included XML --><include file="--.xml"/>
</member>
```

**Comment by davkean** [comment]
That looks like a bug, while "-->" is probably invalid, "--" is not.

**Comment by tannergooding** [comment]
@davke, '--' is invalid. https://www.w3.org/TR/2008/REC-xml-20081126/

> 2.5 Comments
>
>[Definition: Comments may appear anywhere in a document outside other markup; in addition, they may appear within the document type declaration at places allowed by the grammar. They are not part of the document's character data; an XML processor may, but need not, make it possible for an application to retrieve the text of comments. **For compatibility, the string " -- " (double-hyphen) must not occur within comments**.] Parameter entity references must not be recognized within comments.

**Comment by cston** [comment]
Fixed in #8899.

**Comment by natidea** [comment]
New Failure:
http://dotnet-ci.cloudapp.net/job/roslyn-internal_stabil_win_dbg_unit32/160/
